DependencyInjection/CompilerPass/EndpointRouterCompilerPass.php

```diff
@@ -95,9 +95,8 @@ private function buildMappingFromRouteDetails($routeDetails, $classToServiceMapp
 
         foreach ($routeDetails as $routeDetail) {
             $controllerKey = $classToServiceMapping[$routeDetail['controller']] ?? $routeDetail['controller'];
-            $separator = isset($classToServiceMapping[$routeDetail['controller']]) ? ':' : '::';
 
-            $mapping[sprintf('%s%s%s', $controllerKey, $separator, $routeDetail['action'])] = $routeDetail['request'];
+            $mapping[sprintf('%s::%s', $controllerKey, $routeDetail['action'])] = $routeDetail['request'];
```

Comment:
Any reason for these changes?

Reply:
yes, because starting symfony 4.1 they added support of `::` and in symfony6+ just `:` doesn't work

Reply:
We change the behavior for the old Symfony versions.

Reply:
yep. I will update composer with minimum symfony version 4.1

Reply:
Stick to `"symfony/routing": "~4.1|~5.0|~6.0"`